chore: update module plugin docs and readme (#3782)
web-infra-dev/modern.js: packages/document/module-doc/docs/en/plugins/official-list/plugin-babel.md

> # Babel plugins

Reviewer: ```suggestion
# Babel Plugin
```

:::tip

> Normally, we don't need to use Babel to convert our code, this plugin is only used as a downgrade.

Reviewer: ```suggestion
Normally, we don't need to use Babel to transpile our code, this plugin is just used as a downgrade.
```

:::

## Quick start

### Install

```bash
# npm
npm install @modern-js/plugin-module-babel -D

# yarn
yarn add @modern-js/plugin-module-babel -D

# pnpm
pnpm add @modern-js/plugin-module-babel -D
```

### Register

You can install the plugin with the following command:

```ts
import moduleTools, { defineConfig } from '@modern-js/module-tools';
import { modulePluginBabel } from '@modern-js/plugin-module-babel';

export default defineConfig({
  plugins: [
    moduleTools(),
    modulePluginBabel(),
  ],
});
```

## Config

> 

Reviewer: Please provide some demo for custom babel options

> see [babel options](https://babeljs.io/docs/options)

Reviewer: ```suggestion
See [babel options](https://babeljs.io/docs/options)
```
